frontend/src/router/routeHandlers.ts

```diff
@@ -137,10 +150,18 @@ export const beforeEachRouteHandler = async (
         return { path: routeItems.dashboard.path };
     }
 
-    // Access privilege guard. This logic might need to be adjusted soon.
+    // Access privilege guard.
     if (to.meta.requiredPrivileges) {
         for (let role of to.meta.requiredPrivileges as Array<string>) {
-            if (!LoginUserState.hasAccessRole(role)) {
+            if (!LoginUserState.hasAccess(role)) {
+                emitRouteToastError(ACCESS_RESTRICTED_ERROR);
+                return { path: routeItems.dashboard.path };
+            }
+            // if require APP_ADMIN role, need to be the admin of the selected application
+            if (
+                role == AdminRoleAuthGroup.AppAdmin &&
+                !LoginUserState.isAdminOfSelectedApplication()
```

Comment:
Hi Nick, I pushed a change regarding the check for "APP_ADMIN". If the router requires "APP_ADMIN", it actually means this router requires user to be the admin of the selected application. So we do a check here `beforeEachRouteHandler`, then no need to add the check in each individual router handler.

Reply:
Sounds good. Thank you so much!